android/app/src/main/java/org/fossasia/openevent/adapters/viewholders/SessionViewHolder.java

```diff
@@ -1,66 +1,307 @@
-    public ImageView locationIcon;
+    ImageView locationIcon;
 
     @BindView(R.id.session_time)
-    public TextView sessionTime;
+    TextView sessionTime;
 
     @BindView(R.id.session_location)
-    public TextView sessionLocation;
+    TextView sessionLocation;
 
     @BindView(R.id.session_bookmark_status)
-    public ImageView sessionBookmarkIcon;
+    ImageView sessionBookmarkIcon;
 
     @BindView(R.id.session_details)
-    public LinearLayout sessionDetailsHolder;
+    LinearLayout sessionDetailsHolder;
 
     @BindView(R.id.session_card)
-    public CardView sessionCard;
+    CardView sessionCard;
 
     @BindView(R.id.titleLinearLayout)
-    public LinearLayout sessionHeader;
+    LinearLayout sessionHeader;
 
     @BindView(R.id.session_status)
-    public TextView sessionStatus;
+    TextView sessionStatus;
+
+    private Session session;
+    private Context context;
+    private OnBookmarkSelectedListener onBookmarkSelectedListener;
 
-    public SessionViewHolder(View itemView) {
+    public static int listPosition;
+    private static final int locationWiseSessionList = 1;
+    private static final int trackWiseSessionList = 4;
+    private static final int speakerWiseSessionList = 2;
+
+    public SessionViewHolder(View itemView, Context context, OnBookmarkSelectedListener onBookmarkSelectedListener) {
         super(itemView);
         ButterKnife.bind(this, itemView);
+        this.context = context;
+        this.onBookmarkSelectedListener = onBookmarkSelectedListener;
+    }
+
+    public void bindSession(int type, int colorInTracks, RealmDataRepository realmRepo) {
+
+        String sessionTitleString = Utils.checkStringEmpty(session.getTitle());
+        String sessionSubTitle = Utils.checkStringEmpty(session.getSubtitle());
+        int color = ContextCompat.getColor(context, R.color.color_primary);
+
+        sessionTitle.setText(sessionTitleString);
+
+        if (Utils.isEmpty(sessionSubTitle)) {
+            sessionSubtitle.setVisibility(View.GONE);
+        } else {
+            sessionSubtitle.setVisibility(View.VISIBLE);
+            sessionSubtitle.setText(sessionSubTitle);
+        }
+
+        sessionStatus.setVisibility(View.GONE);
+
+        setSessionStatus();
+
+        Track track = session.getTrack();
+
+        if (!RealmDataRepository.isNull(track)) {
+            int storedColor = Color.parseColor(track.getColor());
+
+            if (type != trackWiseSessionList) {
+                color = storedColor;
+            } else {
+                color = colorInTracks;
+            }
+
+            TextDrawable drawable = OpenEventApp.getTextDrawableBuilder().round()
+                    .build(String.valueOf(track.getName().charAt(0)), storedColor);
+
+            trackImageIcon.setImageDrawable(drawable);
+            trackImageIcon.setBackgroundColor(Color.TRANSPARENT);
+            sessionTrack.setText(track.getName());
+
+
+            itemView.setOnClickListener(v -> {
```

Comment:
Best would be to transfer control back to fragment if an item is clicked

Reply:
Ok I will pass the session object through the interface method or should I pass the session properties that I need?